dbt/include/spark/macros/materializations/clone.sql

```diff
@@ -0,0 +1,76 @@
+{% macro spark__can_clone_table() %}
+    {{ return(True) }}
+{% endmacro %}
+
+{% macro spark__create_or_replace_clone(this_relation, defer_relation) %}
+    create or replace table {{ this_relation }} shallow clone {{ defer_relation }}
+{% endmacro %}
+
+{%- materialization clone, adapter='spark' -%}
+
+  {%- set relations = {'relations': []} -%}
+
+  {%- if not defer_relation -%}
+      -- nothing to do
+      {{ log("No relation found in state manifest for " ~ model.unique_id, info=True) }}
+      {{ return(relations) }}
+  {%- endif -%}
+
+  {%- set existing_relation = load_cached_relation(this) -%}
+
+  {%- if existing_relation and not flags.FULL_REFRESH -%}
+      -- noop!
+      {{ log("Relation " ~ existing_relation ~ " already exists", info=True) }}
+      {{ return(relations) }}
+  {%- endif -%}
+
+  {%- set other_existing_relation = load_cached_relation(defer_relation) -%}
+  {%- set file_format = config.get('file_format', validator=validation.any[basestring]) -%}
+
+  -- If this is a database that can do zero-copy cloning of tables, and the other relation is a table, then this will be a table
+  -- Otherwise, this will be a view
+
+  {% set can_clone_table = can_clone_table() %}
+
+  {%- if file_format != 'delta' -%}
+    {% set invalid_format_msg -%}
+      Invalid file format: {{ file_format }}
+      shallow clone requires file_format be set to 'delta'
```

Comment:
curious, why can't we do this with non-delta formats?

Reply:
this may be a misunderstanding on my part honestly will double check I know are writeup and the docs for these specify `delta tables` might of confused that with our delta format suggestion we use in some other places